test/tests.cpp

```diff
@@ -246,38 +242,38 @@ Print all primes between 1 and n
   auto tokenizer = model->CreateTokenizer();
   auto tokens = tokenizer->Encode(prompt);
 
-  Generators::SearchParams params{*model};
+  Generators::GeneratorParams params{*model};
   params.batch_size = 1;
   params.sequence_length = static_cast<int>(tokens.size());
   params.input_ids = tokens;
   params.max_length = 128;
 
+  // TODO: Not sure if we want to keep this "original version" artifact
   // Original version
-  auto search = params.CreateSearch();
-  auto state = model->CreateState(search->GetSequenceLengths(), params);
+  // auto search = params.CreateSearch();
+  // auto state = model->CreateState(search->GetSequenceLengths(), params);
 
-  while (!search->IsDone()) {
-    search->SetLogits(state->Run(search->GetSequenceLength(), search->GetNextTokens()));
-    search->SelectTop();
-  }
+  // while (!search->IsDone()) {
+  //   search->SetLogits(state->Run(search->GetSequenceLength(), search->GetNextTokens()));
+  //   search->SelectTop();
+  // }
 
-  auto result = search->GetSequence(0);
+  // auto result = search->GetSequence(0);
 
   // Generator version
-  auto generator = model->CreateGenerator();
+  auto generator = Generators::CreateGenerator(*model, params);
   while (!generator->IsDone()) {
-    auto logits = generator->RunStep();
-
-    generator->SelectTop();
+    generator->ComputeLogits();
+    generator->GenerateNextToken_Top();
   }
 
   auto result = generator->GetSequence(0);
 
+  // TODO: Not sure if we want this either
   // High level version
-  auto result = model->Generate(params);
+  // auto result = Generators::Generate(model, params);
```

Comment:
We do want this - but it should be in a separate test to show the high level API end to end